### DISCUSSION TITLE
Newbie Button Help

### DISCUSSION THREAD

**Discussion by ValkyrX** [foxglove/community#777] [Q&A]
Just starting with foxglove,so far so good, i am using it with a turtlebot4 , i would like to have a button that says dock / undock etc, so when i press it the bot will dock etc, how would i do this ? i have th ecommand to dock "ros2 action send_goal /dock irobot_create_msgs/action/Dock "{}" how do i put this under a puch button please ?

Many Thanks

**Comment by amacneil** [comment]
We do not yet have built in UI to trigger ROS services or actions. So to do this you would need to create a [custom panel](https://foxglove.dev/docs/studio/extensions/guide-create-custom-panel) where you can make a button trigger [`callService`](https://foxglove.dev/docs/studio/extensions/panel-api#callservice)

**Comment by ValkyrX** [comment]
Ok , so i have managed to build the call service panel and it loads ok, so if the command i need to pass is

ros2 action send_goal /dock irobot_create_msgs/action/Dock

what do i put in service name and request ? tried a few different string but i just get errors

**Reply by achim-k** [reply]
There is currently no builtin support for calling ROS2 actions. You will have to use the hidden services for that.

Below is an example for sending a goal to an action using the [call-service](https://github.com/foxglove/create-foxglove-extension/tree/main/examples/call-service-panel-example) example panel:

![image](https://user-images.githubusercontent.com/9250155/231853814-06533812-4c1f-449c-9d9a-68885ad54ebb.png)

Note that when using foxglove_bridge, you will have to set the `include_hidden` parameter for that to work.

**Reply by jtbandes** [reply]
Just an update that we now have a built in service call panel: https://docs.foxglove.dev/docs/visualization/panels/service-call
